drivers/spi/spi_nrfx_spim.c

```diff
@@ -366,11 +366,17 @@ static int spim_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 }
 #endif /* CONFIG_DEVICE_POWER_MANAGEMENT */
 
-#define SPIM_NRFX_MISO_PULL_DOWN(idx) \
-	IS_ENABLED(DT_NORDIC_NRF_SPIM_SPI_##idx##_MISO_PULL_DOWN)
+/*
+ * We use NODELABEL here because the nrfx HAL API requires us to call
```

Comment:
HAL can be misleading in the context of nrfx, because there we describe this way a layer for accessing peripheral registers via inline functions. And the functions for which we need the SoC peripheral instance index are in nrfx drivers API. I'd suggest to use just "nrfx API" instead.